src/ATen/NativeFunctions.h

```diff
@@ -98,5 +99,29 @@ static inline Tensor permute(const Tensor & self, IntList dims) {
   return self.as_strided(newSizes, newStrides);
 }
 
+/*
+[NativeFunction]
+name: expand
+arg: Tensor self
+arg: IntList sizes
+return: Tensor
+variants: method, function
+type_method_definition_level: base
+type_method_definition_dispatch: at::native::expand
```

Comment:
What do these `type_method_definition_*` things mean?

Reply:
type_method_definition_level: base just indicates that the definition of this function goes into the base Type; we don't need derived-type specific overloads.

type_method_definition_dispatch just gives the name that is dispatched to in the above type definition.  This is just to avoid encoding some magic like all native functions must be in at::native.